docs/bugs_and_glitches.md

```diff
@@ -122,6 +122,7 @@ Fixes in the [multi-player battle engine](#multi-player-battle-engine) category
 +	jp nz, HasEnemyFainted
 +	ret
 +
+-; BUG: Perish Song and Spikes can leave a Pokemon with 0 HP and not faint (see docs/bugs_and_glitches.md)
```

Comment:
Would it make sense to have an identical BUG comment above `CheckFaint_EnemyThenPlayer`?

Reply:
I removed a lot of the identical BUG comments during my most recent changes. I tried to put the BUG comments near the beginning of the actual code changes. In my opinion, unless the changes traverse multiple files or have two distinct purposes like \`Counter and Mirror Coat still work if the opponent uses an item\` with Counter & Mirror Coat. In this case  `CheckFaint_EnemyThenPlayer` exists directly after `CheckFaint_PlayerThenEnemy`, so would just be redundant.

My intent with my second pass of changes, was that the BUG comments serve as markers on where to start applying changes.